app/com/linkedin/drelephant/mapreduce/MapReduceFetcherHadoop2.java

```diff
@@ -133,8 +174,131 @@ public MapReduceApplicationData fetchData(AnalyticJob analyticJob) throws IOExce
     return jobData;
   }
 
-  private String parseException(String jobId, String diagnosticInfo) throws MalformedURLException, IOException,
-                                                                            AuthenticationException {
+  /**
+   * Fetch information for running jobs. When job is in Running state,
+   * check for all the completed tasks and compute the metrics.
+   *
+   * @param analyticJob the job being analyzed
+   * @return
+   * @throws IOException
+   * @throws AuthenticationException
+   * @throws InterruptedException
+   */
+  private MapReduceApplicationData fetchRunningJobsData(AnalyticJob analyticJob, Job job)
+      throws IOException, AuthenticationException, InterruptedException {
+    if (job == null) {
+      return null;
+    }
+
+    MapReduceApplicationData jobData = new MapReduceApplicationData();
+    String appId = analyticJob.getAppId();
+    String jobId = Utils.getJobIdFromApplicationId(appId);
+    String status = analyticJob.getJobStatus();
+    jobData.setAppId(appId).setJobId(jobId).setStatus(status);
+
+    List<TaskReport> mapTaskReportList = Arrays.asList(job.getTaskReports(TaskType.MAP));
+    List<TaskReport> redTaskReportList = Arrays.asList(job.getTaskReports(TaskType.REDUCE));
+
+    int mapSampleSize = getSampled(mapTaskReportList);
+    int redSampleSize = getSampled(redTaskReportList);
+
+    // Fetch task data
+    List<MapReduceTaskData> mapperList = getMRTaskData(mapTaskReportList, mapSampleSize);
+    List<MapReduceTaskData> reducerList = getMRTaskData(redTaskReportList, redSampleSize);
+
+    // Capture the job configuration
+    Properties props = new Properties();
+    if (job.getConfiguration() != null) {
```

Comment:
Since this doesn't change, you can get it once and keep it instead of fetching it every min